### DISCUSSION TITLE
In-memory Sqlite database dropped?

### DISCUSSION THREAD

**Discussion by aeblyve** [rwf2/Rocket#2541] [Questions] [answered]
I'm working on an application which uses a `database` from `rocket_sync_db_pools`. Something like:

```rust
#[database("memory_database")]
pub struct MemoryDatabase(diesel::SqliteConnection);
```

Where memory_database has url `:memory:`.

I'm using it with request guards like:

```
#[post("/edit/pages/<path..>", data = "<new_page>")]
pub async fn edit_page(
    state: &State<ManagedState>,
    new_page: Form<PageInfo>,
    path: PathBuf,
    _admin: AuthenticatedAdmin,
    connection: PersistDatabase,
    memory_connection: MemoryDatabase,
```

At a seemingly random time into running, a table created in this database no longer exists, leading me to believe the in-memory database is dropped or re-connected somehow.

Is this a feature? Is there a way to make the database persist for a whole run?

**Comment by SergioBenitez** [comment, marked as answer]
This is expected and is unrelated to Rocket, if you're using just `:memory:`, which is almost certainly not what you want. See the [SQLite docs](https://www.sqlite.org/inmemorydb.html). You probably want to use a shared, named cache with your in-memory database.

**Reply by aeblyve** [reply]
That's very helpful, thank you.
